### PR TITLE
BUG: TimedeltaArray.astype(duration[{unit}][pyarrow])

### PR DESCRIPTION
**Pull request by lukemanley**
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/v2.0.0.rst` file if fixing a bug or adding a new feature.

Fixes the following:

```
import pandas as pd

arr = pd.array([pd.Timedelta(days=1)]) 
arr.astype("duration[ns][pyarrow]")     # <- raises
```

`TypeError: Cannot convert ArrowDtype to numpy.dtype`

### CODE REVIEW

File: pandas/tests/extension/test_arrow.py [hunk 1]
```diff
@@ -1369,3 +1369,12 @@ def test_pickle_roundtrip(data):
 
     result_sliced = pickle.loads(sliced_pickled)
     tm.assert_series_equal(result_sliced, expected_sliced)
+
+
+def test_astype_from_non_pyarrow(data):
```
Comment (jbrockmendel) [inline pandas/tests/extension/test_arrow.py:1374]: should this go in the TimedeltaArray tests?

File: pandas/tests/extension/test_arrow.py [hunk 2]
```diff
@@ -1369,3 +1369,12 @@ def test_pickle_roundtrip(data):
...
+    # GH49795
+    pd_array = data._data.to_pandas().array
```
Comment (jbrockmendel) [inline pandas/tests/extension/test_arrow.py:1376]: assert that this is a TimedeltaArray?

Comment (lukemanley) [reply pandas/tests/extension/test_arrow.py:1376]: This test is not timedelta specific. It tests all of the types in the `data` fixture. Timedelta was the only one that raised with `{non-pyarrow array}.astype({ArrowDtype})`.

**Review by jbrockmendel — APPROVED** [review]

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @lukemanley

**Pull request merged**